org.eclipse.scanning.example/src/org/eclipse/scanning/example/malcolm/DummyMalcolmRecord.java

```diff
@@ -0,0 +1,453 @@
+package org.eclipse.scanning.example.malcolm;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.epics.pvaccess.server.rpc.RPCResponseCallback;
+import org.epics.pvaccess.server.rpc.RPCServiceAsync;
+import org.epics.pvaccess.server.rpc.Service;
+import org.epics.pvdata.factory.FieldFactory;
+import org.epics.pvdata.factory.PVDataFactory;
+import org.epics.pvdata.factory.StatusFactory;
+import org.epics.pvdata.pv.FieldBuilder;
+import org.epics.pvdata.pv.FieldCreate;
+import org.epics.pvdata.pv.PVBoolean;
+import org.epics.pvdata.pv.PVBooleanArray;
+import org.epics.pvdata.pv.PVDataCreate;
+import org.epics.pvdata.pv.PVDoubleArray;
+import org.epics.pvdata.pv.PVFloat;
+import org.epics.pvdata.pv.PVFloatArray;
+import org.epics.pvdata.pv.PVInt;
+import org.epics.pvdata.pv.PVIntArray;
+import org.epics.pvdata.pv.PVString;
+import org.epics.pvdata.pv.PVStringArray;
+import org.epics.pvdata.pv.PVStructure;
+import org.epics.pvdata.pv.PVUnion;
+import org.epics.pvdata.pv.ScalarType;
+import org.epics.pvdata.pv.Status;
+import org.epics.pvdata.pv.Status.StatusType;
+import org.epics.pvdata.pv.Structure;
+import org.epics.pvdata.pv.Union;
+import org.epics.pvdatabase.PVDatabase;
+import org.epics.pvdatabase.PVDatabaseFactory;
+import org.epics.pvdatabase.PVRecord;
+
+class DummyMalcolmRecord extends PVRecord {
+	
+    private static final FieldCreate fieldCreate = FieldFactory.getFieldCreate();
+    private static final PVDataCreate pvDataCreate = PVDataFactory.getPVDataCreate();
+
+    private boolean     underControl = false;
+    
+    private Map<String, PVStructure> receivedRPCCalls = new HashMap<String, PVStructure>();
+
+	public Map<String, PVStructure> getReceivedRPCCalls() {
+		return receivedRPCCalls;
+	}
+
+	synchronized boolean takeControl() {
+        if (!underControl) {
+            underControl = true;
+            return true;
+        }
+        return false;
+    }
+
+    synchronized void releaseControl() {
+        underControl = false;
+    }
+
+    private class RPCServiceAsyncImpl implements RPCServiceAsync {
+
+        private DummyMalcolmRecord pvRecord;
+        private final Status statusOk = StatusFactory.
+                getStatusCreate().getStatusOK();
+        private String methodName = "";
+
+        RPCServiceAsyncImpl(DummyMalcolmRecord record, String methodName) {
+            pvRecord = record;
+            this.methodName = methodName;
+        }
+
+        public void request(PVStructure args, RPCResponseCallback callback)
+        {
+        	System.out.println("Got Async Request:");
+        	System.out.println(args.toString());

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS106)